src/nvim/buffer_defs.h

```diff
@@ -1046,6 +1048,12 @@ typedef enum {
   kWinStyleMinimal,  /// Minimal UI: no number column, eob markers, etc
 } WinStyle;
 
+typedef enum {
+  kBorderTitleLeft   = 0,
+  kBorderTitleCenter = 1,
+  kBorderTitleRight  = 2,
+} BorderTitlePos;
```

Comment:
there is a lot of special-case code in this PR. Let's try to re-use concepts where possible. "left/right/center/..." is not a unique concept. Is there not already an enum similar to this? If not, maybe a generic `kAlignLeft`/`kAlignCenter`'/... makes sense.